src/mlpack/core/data/split_data.hpp

```diff
@@ -224,6 +228,122 @@ Split(const arma::Mat<T>& input,
                          std::move(testData));
 }
 
+template <class fieldType,
+          class = std::enable_if_t<
+              arma::is_Col<typename fieldType::object_type>::value ||
+              arma::is_Mat_only<typename fieldType::object_type>::value>>
+void Split(fieldType& input, 
+           fieldType& inputLabel,
+           fieldType& trainData,
+           fieldType& trainLabels,
+           fieldType& testData, 
+           fieldType& testLabels,
+           const double testRatio, 
+           const bool shuffleData = true) 
+{
+  const size_t testSize = static_cast<size_t>(input.n_cols * testRatio);
+  const size_t trainSize = input.n_cols - testSize;
+
+  arma::uvec order = arma::linspace<arma::uvec>(0, input.n_cols - 1,
+      input.n_cols);
+  if (shuffleData)
+    order = arma::shuffle(order);
+
+  if (trainSize > 0)
+  {
+    trainLabels.set_size(1, trainSize);
```

Comment:
If `trainSize` is `0`, we should be sure to set the size of `trainLabels` *and* `trainData` accordingly.  In fact, I would suggest calls to `set_size()` for all the output objects just after you compute the `testSize` and `trainSize`. :+1: